home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672
    throw new Error(`Failed to connect to database: ${error} - DB URI ${uri}`);
          ^

Error: Failed to connect to database: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/ - DB URI mongodb://SIP-Dev-SystemAccount:pass@sip-dev-pl-0.eovja.mongodb.net/sip?connectTimeoutMS=15000&retryWrites=true&w=majority
    at connectDatabase (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672:11)

Node.js v18.19.1
/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/node_modules/sequelize/lib/dialects/postgres/connection-manager.js:133
                reject(new sequelizeErrors.ConnectionRefusedError(err));
                       ^

ConnectionRefusedError [SequelizeConnectionRefusedError]: connect ECONNREFUSED ::1:5432
    at Client._connectionCallback (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/node_modules/sequelize/src/dialects/postgres/connection-manager.js:184:24)
    at Client._handleErrorWhileConnecting (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/node_modules/pg/lib/client.js:326:19)
    at Client._handleErrorEvent (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/node_modules/pg/lib/client.js:336:19)
    at Connection.emit (node:events:517:28)
    at Socket.reportStreamError (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/node_modules/pg/lib/connection.js:57:12)
    at Socket.emit (node:events:517:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21) {
  parent: Error: connect ECONNREFUSED ::1:5432
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 5432
  },
  original: Error: connect ECONNREFUSED ::1:5432
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 5432
  }
}

Node.js v18.19.1
FAIL src/modules/product/_test_/product.routes.spec.ts
  ● Test suite failed to run

    Jest worker encountered 4 child process exceptions, exceeding retry limit

      at ChildProcessWorker.initialize (../node_modules/jest-worker/build/workers/ChildProcessWorker.js:181:21)

/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672
    throw new Error(`Failed to connect to database: ${error} - DB URI ${uri}`);
          ^

Error: Failed to connect to database: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/ - DB URI mongodb://SIP-Dev-SystemAccount:pass@sip-dev-pl-0.eovja.mongodb.net/sip?connectTimeoutMS=15000&retryWrites=true&w=majority
    at connectDatabase (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672:11)

Node.js v18.19.1
/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672
    throw new Error(`Failed to connect to database: ${error} - DB URI ${uri}`);
          ^

Error: Failed to connect to database: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/ - DB URI mongodb://SIP-Dev-SystemAccount:pass@sip-dev-pl-0.eovja.mongodb.net/sip?connectTimeoutMS=15000&retryWrites=true&w=majority
    at connectDatabase (/home/jenkins/agent/workspace/FUX_website_-_rdx-sip-bs-ms_prod/src/lib/db.connection.ts:672:11)

above error i am getting while running test cases how to resolve this?following is my test file. 
"""
import { mockRequest, mockResponse } from "../../../utils/MockInterceptor.js";
import {ProductMock} from "../../../mocks/Product.mock.js";
import ProductModel, { initializeProductModel } from "../product.model.js";
import productRouter from "../product.routes.js";
import CalculationsModel, { initializeCalculationModel } from "../../calculation_api/calculation.model.js";
import FormulaModel, { initializeFormulaModel } from "../../formula/formula.model.js";
import FormulaMock from "../../../mocks/Formula.mock.json";
import ComponentSearchMock from "../../../mocks/ComponentSeach.mock.json";
import initializeProductController from "../product.controller.js";
import RawMaterialsModel, { initializeRawMaterialModel } from "../../raw_material_search/rm_search.model.js";
import RawMaterialMock from "../../../mocks/RawMaterial.mock.json";
import ProductSearchModel, { initializeComponentSearchModel } from "../../component_pc_search/component_search.model.js";
import express from "express";
jest.mock('sequelize', () => {
  const actualSequelize = jest.requireActual('sequelize');
  return {
    ...actualSequelize,
    Sequelize: jest.fn(() => ({
      authenticate: jest.fn().mockResolvedValueOnce(null),
      define: jest.fn().mockImplementation((modelName) => modelName),
      sync: jest.fn().mockResolvedValueOnce(null),
      close: jest.fn().mockResolvedValueOnce(null),
    })),
  };
});

jest.mock('../../../lib/db.connection', () => ({
  connectDatabase: jest.fn().mockResolvedValue({
    connection: { readyState: 1 },
    disconnect: jest.fn()
  })
}));
const mockedProductModel = ProductModel as jest.Mock;
const mockedinitializeProductModel = initializeProductModel as jest.Mock;
const mockedCalculationsModel = CalculationsModel as jest.Mock;
const mockedinitializeCalculationModel = initializeCalculationModel as jest.Mock;
const mockedFormulaModel = FormulaModel as jest.Mock;
const mockedinitializeFormulaModel = initializeFormulaModel as jest.Mock;
const mockedRawMaterialsModel = RawMaterialsModel as jest.Mock;
const mockedinitializeRawMaterialModel = initializeRawMaterialModel as jest.Mock;
const mockedProductSearchModel = ProductSearchModel as jest.Mock;
const mockedinitializeComponentSearchModel = initializeComponentSearchModel as jest.Mock;

jest.mock("p-limit", () => {
  return () => 12;
});


jest.mock("../product.model", () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => ({
    aggregate: jest.fn().mockReturnThis(),
    findOne: jest.fn().mockResolvedValue({ ...ProductMock[0] }),
    create: jest.fn().mockResolvedValue({ ...ProductMock[0] }),
    // Mock other methods as needed
  })),
  initializeProductModel: jest.fn(),
}));
jest.mock("../../calculation_api/calculation.model");
jest.mock("../../formula/formula.model");
jest.mock("../../raw_material_search/rm_search.model");
jest.mock("../../component_pc_search/component_search.model");
"""
how to mock sequelize for other model.
is my ../product.model" is correct? do for all to resolve the error .
because we cant whitelist the mongodb and postgre diresctly we have to fix in test file only . help me
